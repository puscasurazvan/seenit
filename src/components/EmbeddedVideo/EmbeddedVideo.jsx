import React from 'react'
import PropTypes from 'prop-types'

/** The video in the iframe is to be shown or hidden by clicking on the heading text */

const EmbeddedVideo = ({ video }) => {
  return (
    <>
      <iframe
        title={video.title}
        src={video.src}
        height="300px"
        width="500px"
      />
    </>
  )
}

EmbeddedVideo.propTypes = {
  video: PropTypes.shape().isRequired,
}

export default EmbeddedVideo
