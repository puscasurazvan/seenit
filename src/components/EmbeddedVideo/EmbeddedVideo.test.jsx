import React from 'react'
import { mount } from 'enzyme'
import EmbeddedVideo from './EmbeddedVideo'

const minProps = {
  video: { title: 'Video Title', src: 'www.url.com' },
}

describe('EmbeddedVideo', () => {
  it('should render with the right props', () => {
    const wrapper = mount(<EmbeddedVideo {...minProps} />)
    expect(wrapper.find('iframe').length).toEqual(1)
    expect(wrapper.props().video).toEqual({
      title: 'Video Title',
      src: 'www.url.com',
    })
    expect(wrapper.find('EmbeddedVideo').length).toEqual(1)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<EmbeddedVideo {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
