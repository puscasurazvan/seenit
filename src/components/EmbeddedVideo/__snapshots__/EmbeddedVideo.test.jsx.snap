// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedVideo should render without exploding 1`] = `ReactWrapper {}`;
